tests/ui/symbol-names/types.rs

```diff
@@ -1,169 +1,270 @@
+    //[v0]~| ERROR ::b::Type<i64>>)
+    //[v0]~| ERROR demangling-alt(<a::b::Type<i64>>)
     impl Type<i64> {}
 
     #[rustc_symbol_name]
-    //~^ ERROR symbol-name(_ZN1a1b14Type$LT$u8$GT$
-    //~| ERROR demangling(a::b::Type<u8>::
-    //~| ERROR demangling-alt(a::b::Type<u8>)
+    //[legacy,verbose-legacy]~^ ERROR symbol-name(_ZN1a1b14Type$LT$u8$GT$
+    //[legacy,verbose-legacy]~| ERROR demangling(a::b::Type<u8>::
+    //[legacy,verbose-legacy]~| ERROR demangling-alt(a::b::Type<u8>)
+    //[v0]~^^^^ ERROR symbol-name(_RMs4_NvCsCRATE_HASH_1a1bINtB<REF>_4TypehE)
+    //[v0]~| ERROR ::b::Type<u8>>)
+    //[v0]~| ERROR demangling-alt(<a::b::Type<u8>>)
     impl Type<u8> {}
 
     #[rustc_symbol_name]
-    //~^ ERROR symbol-name(_ZN1a1b15Type$LT$u16$GT$
-    //~| ERROR demangling(a::b::Type<u16>::
-    //~| ERROR demangling-alt(a::b::Type<u16>)
+    //[legacy,verbose-legacy]~^ ERROR symbol-name(_ZN1a1b15Type$LT$u16$GT$
+    //[legacy,verbose-legacy]~| ERROR demangling(a::b::Type<u16>::
+    //[legacy,verbose-legacy]~| ERROR demangling-alt(a::b::Type<u16>)
+    //[v0]~^^^^ ERROR symbol-name(_RMs5_NvCsCRATE_HASH_1a1bINtB<REF>_4TypetE)
+    //[v0]~| ERROR ::b::Type<u16>>)
+    //[v0]~| ERROR demangling-alt(<a::b::Type<u16>>)
     impl Type<u16> {}
 
     #[rustc_symbol_name]
-    //~^ ERROR symbol-name(_ZN1a1b15Type$LT$u32$GT$
-    //~| ERROR demangling(a::b::Type<u32>::
-    //~| ERROR demangling-alt(a::b::Type<u32>)
+    //[legacy,verbose-legacy]~^ ERROR symbol-name(_ZN1a1b15Type$LT$u32$GT$
+    //[legacy,verbose-legacy]~| ERROR demangling(a::b::Type<u32>::
+    //[legacy,verbose-legacy]~| ERROR demangling-alt(a::b::Type<u32>)
+    //[v0]~^^^^ ERROR symbol-name(_RMs6_NvCsCRATE_HASH_1a1bINtB<REF>_4TypemE)
+    //[v0]~| ERROR ::b::Type<u32>>)
+    //[v0]~| ERROR demangling-alt(<a::b::Type<u32>>)
     impl Type<u32> {}
 
     #[rustc_symbol_name]
-    //~^ ERROR symbol-name(_ZN1a1b15Type$LT$u64$GT$
-    //~| ERROR demangling(a::b::Type<u64>::
-    //~| ERROR demangling-alt(a::b::Type<u64>)
+    //[legacy,verbose-legacy]~^ ERROR symbol-name(_ZN1a1b15Type$LT$u64$GT$
+    //[legacy,verbose-legacy]~| ERROR demangling(a::b::Type<u64>::
+    //[legacy,verbose-legacy]~| ERROR demangling-alt(a::b::Type<u64>)
+    //[v0]~^^^^ ERROR symbol-name(_RMs7_NvCsCRATE_HASH_1a1bINtB<REF>_4TypeyE)
+    //[v0]~| ERROR ::b::Type<u64>>)
+    //[v0]~| ERROR demangling-alt(<a::b::Type<u64>>)
     impl Type<u64> {}
 
     #[rustc_symbol_name]
-    //~^ ERROR symbol-name(_ZN1a1b15Type$LT$f32$GT$
-    //~| ERROR demangling(a::b::Type<f32>::
-    //~| ERROR demangling-alt(a::b::Type<f32>)
+    //[legacy,verbose-legacy]~^ ERROR symbol-name(_ZN1a1b15Type$LT$f16$GT$
+    //[legacy,verbose-legacy]~| ERROR demangling(a::b::Type<f16>::
+    //[legacy,verbose-legacy]~| ERROR demangling-alt(a::b::Type<f16>)
+    //[v0]~^^^^ ERROR symbol-name(_RMs8_NvCsCRATE_HASH_1a1bINtB<REF>_4TypeC3f16E)
+    //[v0]~| ERROR ::b::Type<f16[0]>>)
```

Comment:
Is there any way to avoid this `[0]`? For `f128` too

Reply:
This looks to me like rustc-demangle sub-optimally handles `C` productions without a disambiguator: https://github.com/rust-lang/rustc-demangle/blob/af38dc6b06b5665c79a4778268e4faf5decd19df/src/v0.rs#L800-L812

It should not treat a missing disambiguator as if it was a `0` disambiguator. Ideally, we would fix rustc-demangle before merging this PR.